addons/html_editor/static/src/others/embedded_components/plugins/syntax_highlighting_plugin/syntax_highlighting_plugin.js

```diff
@@ -97,15 +69,11 @@ export class SyntaxHighlightingPlugin extends Plugin {
             const isPreInSelection = !targetedNodes.some((node) => !pre.contains(node));
             const codeBlock = renderToElement("html_editor.EmbeddedSyntaxHighlightingBlueprint", {
                 embeddedProps: JSON.stringify({
-                    initialValue: this.getPreValue(pre),
+                    value: getPreValue(pre),
+                    languageId: pre.dataset.languageId || DEFAULT_LANGUAGE_ID,
                     autofocus: preserveFocus && isPreInSelection,
```

Comment:
with my PR suggestion, we will be able to do `embeddedInsertProps: JSON.stringify({ autofocus: preserveFocus && isPreInSelection }), (and adapt the blueprint template)